arcaflow_plugin_stressng/stressng_schema.py

```diff
@@ -203,318 +558,302 @@ class StressNGParams:
                 ],
                 typing.Annotated[
                     HDDStressorParams,
                     annotations.discriminator_value("hdd"),
                     schema.name("HDD Stressor Parameters"),
-                    schema.description(
-                        "Parameters for running the hdd stressor"
-                    ),
+                    schema.description("Parameters for running the hdd stressor"),
                 ],
             ],
-            annotations.discriminator("stressor"),
+            annotations.discriminator("stressor", discriminator_inlined=True),
             schema.name("Stressors List"),
             schema.description("List of stress-ng stressors and parameters"),
         ]
     ]
-    verbose: typing.Optional[bool] = field(
-        default=None,
-        metadata={"name": "verbose", "description": "verbose output"},
-    )
-    metrics_brief: typing.Optional[bool] = field(
-        default=None,
-        metadata={
-            "name": "brief metrics",
-            "description": "Brief version of the metrics output",
-        },
-    )
-
-    workdir: typing.Optional[str] = field(
-        default=None,
-        metadata={
-            "name": "Working Dir",
-            "description": (
-                "Path were stress-ng will be "
-                "executed (example to target a specific volume)"
-            ),
-        },
-    )
 
-    def to_jobfile(self) -> str:
-        result = "timeout {}\n".format(self.timeout)
-        if self.verbose is not None:
-            result = result + "verbose {}\n".format(self.verbose)
-        if self.metrics_brief is not None:
-            result = result + "metrics-brief {}\n".format(self.metrics_brief)
-        return result
+    verbose: typing.Annotated[
+        typing.Optional[bool],
+        schema.name("Verbose"),
+        schema.description("verbose output"),
+    ] = None
 
+    metrics_brief: typing.Annotated[
+        typing.Optional[bool],
+        schema.name("Brief Metrics"),
+        schema.description("Brief version of the metrics output"),
+    ] = None
 
-@dataclass
-class WorkloadParams:
-    StressNGParams: typing.Annotated[
-        StressNGParams,
-        schema.name("Stress-NG Job Parameters"),
+    workdir: typing.Annotated[
+        typing.Optional[str],
+        schema.name("Working Dir"),
         schema.description(
-            """
-            Global workload parameters and list of stressors
-            for the stress-ng job
-            """
+            "Path were stress-ng will be "
+            "executed (example to target a specific volume)"
         ),
-    ]
+    ] = None
+
     cleanup: typing.Annotated[
         typing.Optional[bool],
         schema.name("Cleanup"),
         schema.description("Cleanup artifacts after the plugin run"),
     ] = False
 
+    def to_jobfile(self) -> str:
+        result = "timeout {}\n".format(self.timeout)
```

Comment:
You changed a lot of the others to f-strings: why not here?

Reply:
If you look at the diffs here on GitHub, there is...well...less.  🙃